lsp-completion.el

```diff
@@ -755,6 +755,20 @@ The CLEANUP-FN will be called to cleanup."
   "Disable LSP completion support."
   (lsp-completion-mode -1))
 
+(defun lsp-completion-passthrough-try-completion (string table pred point)
+  (let* ((completion-ignore-case t)
```

Comment:
AFAICS lsp-mode doesn't use this variable when matching: the list of completions is constructed case-insensitively, at least with the language server that I'm currently testing with.

Reply:
Ah, that's right. `lsp-mode` is using `string-match`, so this should be bound to `case-fold-search` instead

Reply:
Okay. Well, when you do string matching in completion, I think you're expected to bind `case-fold-search` to `completion-ignore-case` around that logic.

Or just go with the value `t`, like I think happens now most of the time.

Reply:
Great, I can bind the `case-fold-search` to `completion-ignore-case` in another PR, or if possible, can you help to bind it in this same PR if possible :).
I think there's only two places in this file.

Reply:
Note that `case-fold-search` is `t` by default, whereas `completion-ignore-case` is nil by default.

Obeying the user-set `completion-ignore-case` would make lsp-mode's completion case-sensitive. This is not bad (Eglot currently does that), but it's a significant change in behavior, I think, and it'll make it less "passthrough".

So, are you sure you want that?

Reply:
I mean we should change that in other places to make it conforms with `completion-ignore-case`. In this passthrough function, just returns `(cons string point)` should be okay. It will be similar to VsCode.
@yyoncho, let me know if you have any concern

Reply:
> In this passthrough function, just returns (cons string point) should be okay.

Like mentioned, I'd like to keep the change in behavior in this PR to a minimum. After the merge, this code would be equivalent to the latest Eglot in Emacs master.

Except Eglot does case matching for prefix (by default) - and lsp-mode ignores case, not sure which is better but both of the corresponding behaviors seem fit the corresponding package' goals.

Reply:
You might very well do that change in the next edit, though (if there is agreement among the maintainers).

Reply:
Got it. I signed off for this change. Let's get another signoff and we can merge